pkg/kusionctl/cmd/compile/compile.go

```diff
@@ -56,20 +56,24 @@ func NewCmdCompile() *cobra.Command {
 		},
 	}
 
-	cmd.Flags().StringSliceVarP(&o.Arguments, "argument", "D", []string{},
-		i18n.T("Specify the top-level argument"))
-	cmd.Flags().StringSliceVarP(&o.Settings, "setting", "Y", []string{},
-		i18n.T("Specify the command line setting files"))
+	AddCompileFlags(cmd, o)
 	cmd.Flags().StringVarP(&o.Output, "output", "o", "",
 		i18n.T("Specify the output file"))
-	cmd.Flags().StringVarP(&o.WorkDir, "workdir", "w", "",
-		i18n.T("Specify the work directory"))
 	cmd.Flags().BoolVarP(&o.DisableNone, "disable-none", "n", false,
 		i18n.T("Disable dumping None values"))
 	cmd.Flags().BoolVarP(&o.OverrideAST, "override-AST", "a", false,
 		i18n.T("Specify the override option"))
-	cmd.Flags().StringSliceVarP(&o.Overrides, "overrides", "O", []string{},
-		i18n.T("Specify the configuration override path and value"))
 
 	return cmd
 }
+
+func AddCompileFlags(cmd *cobra.Command, o *CompileOptions) {
+	cmd.Flags().StringVarP(&o.WorkDir, "workdir", "w", "",
+		i18n.T("Specify the work directory"))
+	cmd.Flags().StringSliceVarP(&o.Settings, "setting", "Y", []string{},
+		i18n.T("Specify the command line setting files"))
+	cmd.Flags().StringArrayVarP(&o.Arguments, "argument", "D", []string{},
+		i18n.T("Specify the top-level argument"))
+	cmd.Flags().StringSliceVarP(&o.Overrides, "overrides", "O", []string{},
+		i18n.T("Specify the configuration override path and value"))
+}
```

Comment:
Consider adding a func to the XXXXOptions:
```go
func (o *CompileOptions) AddFlags(cmd *cobra.Command) {
	cmd.Flags().StringVarP(&o.WorkDir, "workdir", "w", "",
		i18n.T("Specify the work directory"))
	cmd.Flags().StringSliceVarP(&o.Settings, "setting", "Y", []string{},
		i18n.T("Specify the command line setting files"))
	cmd.Flags().StringArrayVarP(&o.Arguments, "argument", "D", []string{},
		i18n.T("Specify the top-level argument"))
	cmd.Flags().StringSliceVarP(&o.Overrides, "overrides", "O", []string{},
		i18n.T("Specify the configuration override path and value"))
}
```
Can refer: https://github.com/kubernetes/component-base/blob/85b8d4a6aaaaf51bbd0dbcf732bc843448abedd0/featuregate/feature_gate.go#L317

Reply:
good advice~